app/screens/edit_profile/edit_profile.tsx

```diff
@@ -0,0 +1,404 @@
+                first_name: userInfo.firstName,
+                last_name: userInfo.lastName,
+                nickname: userInfo.nickname,
+                position: userInfo.position,
+                username: userInfo.username,
+            };
+
+            const {error: reqError} = await updateMe(serverUrl, partialUser);
+
+            if (reqError) {
+                resetScreen(reqError as Error);
+                return;
+            }
+
+            close();
+        } catch (e) {
+            resetScreen(e as Error);
+        }
+    });
+
+    const resetScreen = (resetError: Error) => {
+        setError(resetError?.message);
+        Keyboard.dismiss();
+        setUpdating(false);
+        enableSaveButton(true);
+        scrollViewRef.current?.scrollToPosition(0, 0, true);
+    };
+
+    const updateField = useCallback((fieldKey: string, name: string) => {
+        const update = {...userInfo};
+        update[fieldKey] = name;
+        setUserInfo(update);
+
+        // @ts-expect-error access object property by string key
+        const currentValue = currentUser[fieldKey];
+        const didChange = currentValue !== name;
+        enableSaveButton(didChange);
+    }, [userInfo]);
+
+    const includesSsoService = (sso: string) => ['gitlab', 'google', 'office365'].includes(sso);
+
+    const fieldSequence: FieldSequence = {
+        firstName: {
+            ref: firstNameRef,
+            isDisabled: lockedFirstName || includesSsoService(service),
+        },
+        lastName: {
+            ref: lastNameRef,
+            isDisabled: lockedLastName || includesSsoService(service),
+        },
+        username: {
+            ref: usernameRef,
+            isDisabled: service !== '',
+        },
+        email: {
+            ref: emailRef,
+            isDisabled: true,
+        },
+        nickname: {
+            ref: nicknameRef,
+            isDisabled: lockedNickname,
+        },
+        position: {
+            ref: positionRef,
+            isDisabled: lockedPosition,
+        },
+    };
+
+    const onFocusNextField = (fieldKey: string) => {
+        const findNextField = () => {
+            const fields = Object.keys(fieldSequence);
+            const curIndex = fields.indexOf(fieldKey);
+            const searchIndex = curIndex + 1;
+
+            if (curIndex === -1 || searchIndex > fields.length) {
+                return undefined;
+            }
+
+            const remainingFields = fields.slice(searchIndex);
+
+            const nextFieldIndex = remainingFields.findIndex((f: string) => {
+                const field = fieldSequence[f];
+                return !field.isDisabled;
+            });
+
+            if (nextFieldIndex === -1) {
+                Keyboard.dismiss();
+                return undefined;
+            }
+
+            const fieldName = remainingFields[nextFieldIndex];
+            return fieldSequence[fieldName];
+        };
+
+        if (fieldKey === 'position' && canSave) {
```

Comment:
really?

Reply:
If we are navigating to the last field and along the way, we are updating some values, then upon pressing `done` on the last `position` field, we do a `keyboard.dimiss()` and submit the user values.

Reply:
If we navigate amongst the fields and edit some values,  this `if` prepares for form submission.

Reply:
Yes I get that, but who's to say that position is the last non disabled field?

Reply:
Corrected